config/routes.rb

```diff
@@ -6,9 +6,10 @@
   resources :shopping_cart_items, only: [:create], as: :items
 
   resources :shopping_carts, only: [:show], as: :cart
+    post "/complete", controller: :shopping_carts, action: :finalize
```

Comment:
Use the same word here to keep things clear - so either `post '/complete', action: :complete` or both as finalize.